backend/src/infra/repository/GroupRepositoryInMemory.ts

```diff
@@ -61,4 +61,19 @@ export class GroupRepositoryInMemory implements GroupRepository {
     this.nextTransactionId += 1;
     return this.save(group);
   }
+
+  async removeTransaction(
+    group: Group,
+    transaction: Transaction
+  ): Promise<Group> {
+    const transactionIndexToRemove = group.transactions.findIndex(
+      (tmpTransaction) => tmpTransaction.id === transaction.id
+    );
+    const updatedTransactions = group.transactions
+      .slice(0, transactionIndexToRemove)
+      .concat(group.transactions.slice(transactionIndexToRemove + 1));
+    group.transactions = updatedTransactions;
```

Comment:
This works, but you can also do it in only one pass using `filtrer`. It is often used with React ^^

```typescript
group.transactions = group.transactions.filter(
  currentTransaction => currentTransaction.id !== transaction.id
);
```

It will keep all transactions with a different ids, hence filtering the one to remove